src/context/QuizContext.tsx

```diff
@@ -0,0 +1,43 @@
+import { createContext, useContext, useEffect, useReducer } from "react";
+import { getCategories, getQuizzes } from "../utils/firebaseHelpers";
+import { quizInitialState, quizReducer } from "../reducers";
+import { QuizContextType, ReactChildren } from "../types";
+import { getErrorMessage } from "../utils";
+
+const QuizContext = createContext({} as QuizContextType);
+
+const QuizProvider = ({ children }: ReactChildren) => {
+  const [quizState, quizDisPatch] = useReducer(quizReducer, quizInitialState);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        quizDisPatch({ type: "INITIALIZE" });
+        const response = await getCategories();
```

Comment:
Can we use keep all these constants in a constants file and reuse them instead writing a string here?